packages/@aws-cdk/cfn-include/lib/cdk-include.ts

```diff
@@ -0,0 +1,221 @@
+import * as core from '@aws-cdk/core';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+/**
+ * Construction properties of {@link CdkInclude}.
+ */
+export interface CdkIncludeProps {
+  /**
+   * Path to the template file.
+   *
+   * Currently, only JSON templates are supported.
+   */
+  readonly templateFile: string;
+
+  /**
+   * Whether the resources should have the same logical IDs in the resulting CDK template
+   * as they did in the original CloudFormation template file.
+   *
+   * @default true
+   */
+  readonly preserveLogicalIds?: boolean;
+}
+
+/**
+ * A construct that allows you to include an existing CloudFormation template file into your CDK application.
+ * You can retrieve any of the resources defined in the template file by calling the {@link getResource} method.
+ * Any modifications you make on the returned resource objects will be reflected in the resulting CDK template.
+ *
+ * @see getResource
+ */
+export class CdkInclude extends core.Construct {
+  private readonly resources: { [logicalId: string]: core.CfnResource } = {};
+  private readonly cfnType2L1Class: { [cfnType: string]: string };
+  private readonly template: any;
+  private readonly preserveLogicalIds: boolean;
+
+  constructor(scope: core.Construct, id: string, props: CdkIncludeProps) {
+    super(scope, id);
+
+    // read the template into a JS object
+    this.template = fs.readJsonSync(props.templateFile);
+
+    // check for any sections of the template we don't recognize
+    for (const section of Object.keys(this.template)) {
+      switch (section) {
+        case 'AWSTemplateFormatVersion':
+        case 'Description':
+        case 'Resources':
+          // currently, we only support these 3 sections
+          break;
+        default:
+          throw new Error(`Encountered an unsupported section in the template: '${section}'`);
+      }
+    }
+
+    // read the map of CloudFormation type to their L1 class created by the build script
+    this.cfnType2L1Class = fs.readJsonSync(path.join(__dirname, '..', 'cfn-types-2-classes.json'));
```

Comment:
Feels like this can be statically loaded once at the module level. It's not instance-specific, right?

Reply:
Generally I would extract all the logic behind working with this mapping into a separate file with a set of static utility methods so this hairy stuff does not mess up this file.

Reply:
Done.